### PR TITLE
[PHPUnitBridge] Add `enum_exists` mock

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Related to https://github.com/symfony/symfony/pull/48516

### CODE REVIEW

File: components/phpunit_bridge.rst [hunk 1]
```diff
@@ -736,6 +736,7 @@ reason, this component also provides mocks for these PHP functions:
 * :phpfunction:`class_exists`
 * :phpfunction:`interface_exists`
 * :phpfunction:`trait_exists`
+* :phpfunction:`enum_exists` (requires PHP 8.1)
```
Comment (OskarStark) [inline components/phpunit_bridge.rst]:
```suggestion
* :phpfunction:`enum_exists`
```

I would remove this hint

cc @javiereguiluz WDYT?

Comment (javiereguiluz) [reply components/phpunit_bridge.rst]: Yes, I think we can remove it because the entire Symfony 6.3 requires >= 8.1.

Comment (alexandre-daubois) [reply components/phpunit_bridge.rst]: Actually, PHP Unit Bridge requires PHP >= 7.1, which is why I put this hint. But I may remove it anyway?

Comment (javiereguiluz) [reply components/phpunit_bridge.rst]: Good catch! Now I think we should keep this explicit mention to the PHP requirement 😄  Thanks

Comment (OskarStark) [reply components/phpunit_bridge.rst]: OK, lets keep it

**Review by OskarStark — APPROVED** [review]

**Comment by ro0NL** [comment]
a combination of withMockedClasses + withMockedEnums should be explained IMHO (https://github.com/symfony/symfony/pull/48516#issuecomment-1353261478)

**Comment by alexandre-daubois** [comment]
@ro0NL the doc was actually written before adding `withMockedEnums`. I think it worth updating this, good idea! I'll have a look

**Comment by ro0NL** [comment]
while at it, we should probably mention mocking a class (symbol) means all *_exists() functions return true

except enums, hence mockWithEnum

**Comment by alexandre-daubois** [comment]
Updated, @javiereguiluz please tell me if I need to rephrase anything! :slightly_smiling_face:

File: components/phpunit_bridge.rst [hunk 2]
```diff
@@ -798,6 +799,13 @@ classes, interfaces and/or traits for the code to run::
         }
     }
 
+Note that mocking a class with ``ClassExistsMock::withMockedClasses()``
+will make :phpfunction:`class_exists`, :phpfunction:`interface_exists`
+and :phpfunction:`trait_exists` return true.
+
+To register an enumeration and mock :phpfunction:`enum_exists`,
+``ClassExistsMock::withMockedEnums()`` must be used.
```
Comment (ro0NL) [inline components/phpunit_bridge.rst]: perhaps it's obvious for ppl, but only yesteday i realized class_exists returns true for enums (https://github.com/symfony/symfony/pull/48516#issuecomment-1353261478)

so what this new method enables doing is:

```php
withMockedClasses([SomeClass::class => true])
withMockedEnums([SomeClass::class => false])
```

to trigger a code path for `if (class_exists(SomeClass::class) && !enum_exists(SomeClass::class)) {`

**Comment by OskarStark** [comment]
@alexandre-daubois can you please check if this PR is up-to-date with the latest work in the code PR? Please ping me afterwards, so I can merge it, thanks

**Comment by alexandre-daubois** [comment]
@OskarStark I updated the documentation! :slightly_smiling_face:

**Comment by OskarStark** [comment]
Thanks Alexandre.

**Pull request merged**